rdrf/rdrf/forms/widgets/settings_widgets.py

```diff
@@ -124,3 +124,55 @@ def render(self, name, value, attrs=None, renderer=None):
             <script>
                 {javascript}
             </script>""")
+
+
+class RadioSelectSettings(Widget):
+    @staticmethod
+    def get_allowed_fields():
+        return {'force_vertical'}
+
+    def generate_input(self, name, title, parsed, info=None):
+        value = parsed.get(name, False)
+
+        input_str = f'''
+            <input type="checkbox" name="{name}" id="{name}" onchange="saveJSON()" {"checked" if value else ""}>
+            '''
+        help_text = f'<div class="help">{info}</div>' if info else ''
+        return f"""
+            <div>
+                <label for="{name}">{title}</label>
+                {input_str}
+                {help_text}
+            </div>"""
+
+    def generate_inputs(self, parsed):
+        rows = [
+            self.generate_input('force_vertical', _('Force vertical layout'), parsed),
+        ]
+        return "<br/>".join(rows)
+
+    def render(self, name, value, attrs=None, renderer=None):
+        parsed = {}
+        try:
+            parsed = json.loads(value)
+        except Exception:
+            pass
+
+        html = """
+             <div style="display:inline-grid" id="id_{name}">
+                {inputs}
+                <input type="hidden" name="{name}" value='{value}'/>
+             </div>""".format(inputs=self.generate_inputs(parsed), name=name, value=value)
+        javascript = """
+            function saveJSON() {
```

Comment:
This looks very similar to saveJSON in TimeWidgetSettings and resembles the one in SliderWidgetSettings. Can you make a base class with some methods to address this ? Probably you can also move generate_input and generate_inputs to that class...

Reply:
Yes, I mentioned this in the original ticket, but I wasn't sure whether we wanted to invest time in refactoring. I think it'd be worth it but I'm not sure how much use this functionality gets, or if there are alternative plans.

If we were to refactor I think it would be best to do a large overhaul and change how the inputs are generated, because at the moment they are only of the same type. We could also make use of actual templates rather than string formatting.

Reply:
These widget settings are new, TRRF code we've added, therefore I think we should invest the time to refactor it now, so let's do it.

There isn't anything stopping us, to release this feature first though and make another refactor task, right?

Reply:
If we create a separate ticket for refactoring then this is good to go.

Reply:
Great.

The other thing I mentioned in the issue is that if we're going to be using a bootstrap grid to layout elements, then perhaps this setting should allow users to choose to force a fixed size between 1-12, rather than just using a size of 12.

Reply:
Let's keep it as is for now.

I think it is good enough and doesn't bring any other new questions to the table, like how to explain the grid system to the end user, use the bootstrap numbering or use a more user-friendlier "2/3/4 items per row" instead.
We also might want to consider these layout type properties in a more generic way for other Widgets as well.